### DISCUSSION TITLE
How to get HTTP response and modify  it in Mirth

### DISCUSSION THREAD

**Discussion by MrTianFei** [nextgenhealthcare/connect#4820] [Q&A] [answered]
step 1：I set the source was "HTTP Listener"
step 2：In source "Edit Transformer", I query database and put the result to "channelMap",
code example:  channelMap.put('patientJson', patient) 
step 3：I set the destination was "HTTP Sender",and made the destination "query parameter" was “patientJson“
![image](https://user-images.githubusercontent.com/86398277/139410998-0dec447f-ff4a-47a6-8ec5-4a9078b1e379.png)
my question:
I now want to modify the response of the HTTP Sender, how do I get the response of the HTTP Sender and modify it in Mirth

**Comment by narupley** [comment, marked as answer]
On your destination settings, click the "Edit Response" task to the left to configure your response transformer. This is similar to a regular transformer, except it transforms your response payload instead of the message. As long as you have at least one step configured and you actually receive a non-blank response payload, then the response transformer will run. If you want to have the response transformer to run all the time, even when there's no actual response payload (like when the message fails to send due to connection problems, etc), you can set your response data types to Raw.

If you want to take that destination response and send it back as the response from your HTTP Listener, you can select that destination from the Response drop-down menu in the Source tab.

Check the user guide for more information on the response transformer as well!

**Reply by MrTianFei** [reply]
Sorry, I just started work today, and I tried your solution. 
May I ask how can I get the response result returned by HTTP Sender through JavaScript in "Edit Response"

**Reply by pacmano1** [reply]
The response is "msg" (meaning the data, e.g. the JSON returned from a call).

However you may want to cal lhttp senders and set the respons to "raw" so that the response transformer always fires and you can inspect the response code / error to then requeue or take other acion.

**Reply by MrTianFei** [reply]
thanks a lot
